hooks/tk-multi-publish2/basic/nuke_update_flame_clip.py

```diff
@@ -570,6 +645,82 @@ def _update_flame_clip(self, item):
             fh.close()
 
 
+def _get_flame_frame_spec_from_path(path):
+    """
+    Parses the file name in an attempt to determine the first and last
+    frame number of a sequence. This assumes some sort of common convention
+    for the file names, where the frame number is an integer at the end of
+    the basename, just ahead of the file extension, such as
+    file.0001.jpg, or file_001.jpg. We also check for input file names with
+    abstracted frame number tokens, such as file.####.jpg, or file.%04d.jpg.
+    Once the start and end frames have been extracted, this is used to build
+    a frame-spec path, such as "/project/foo.[0001-0010].jpg".
+
+    :param str path: The file path to parse.
+
+    :returns: If the path can be parsed, a string path replacing the frame
+        number with a frame range spec is returned.
+    :rtype: str or None
+    """
+    # This pattern will match the following at the end of a string and
+    # retain the frame number or frame token as group(1) in the resulting
+    # match object:
+    #
+    # 0001
+    # ####
+    # %04d
+    #
+    # The number of digits or hashes does not matter; we match as many as
+    # exist.
+    frame_pattern = re.compile(r"^(.+?)([0-9#]+|[%]0\dd)$")
+    root, ext = os.path.splitext(path)
+
+    # NOTE:
+    #   match.group(0) is the entire path.
+    #   match.group(1) is everything up to the frame number.
+    #   match.group(2) is the frame number.
+    match = re.search(frame_pattern, root)
+
+    # If we did not match, we don't know how to parse the file name, or there
+    # is no frame number to extract.
+    if not match:
+        return None
+
+    # We need to get all files that match the pattern from disk so that we
+    # can determine what the min and max frame number is. We replace the
+    # frame number or token with a * wildcard.
+    glob_path = "%s%s" % (
+        re.sub(match.group(2), "*", root),
+        ext,
+    )
+    files = glob.glob(glob_path)
```

Comment:
@josh-t I realize now that this might be overkill. Is it safe to use the sequence_paths property from the publish item? Is it always going to include the entire list of frames on disk?

Reply:
Currently I don't think so. I took a look and I think this is true for sg writenode-based items and items created via folder drop. But it doesn't look to be the case with regular write nodes. I need to go back and take a closer look and figure out why we're not building that list for those items. That said, you might be able to take advantage of [this](http://developer.shotgunsoftware.com/tk-multi-publish2/utility.html#tk_multi_publish2.util.get_frame_sequences) method here instead of writing your own logic.